doc/tools/docgram/orderedGrammar

```diff
@@ -2379,9 +2379,9 @@ ltac2_tactic_atom: [
 
 ltac2_quotations: [
 | "ident" ":" "(" lident ")"
+| "pat" ":" "(" cpattern ")"
```

Comment:
FWIW, I try to keep updated lines in the same order for ease of reading changes and to have less change in the code and doc.  If you run `make doc_gram_rsts` with these change, it will change the order in `ltac2.rst`.

Reply:
This change was auto-generated using `dune @check-gram`, so it means that I should once more update the `dune` rules to match a change in the `Makefile` logic.